### ISSUE TITLE
Cannot find reader.mat when in executable form

### ISSUE THREAD

**Issue by DoNoRe** [blink1073/oct2py#144]
Hi! 
I am trying to make a python executable which uses oct2py to eval a .m file. The python script works just fine independently. I have been successful in using pyinstaller to make an executable (using --onefile, no errors in building it), but when I run this executable I get the following error and the executable quits:

```
    C:\Users\Jason\Desktop\Oct2Py test>readFolder.exe
    error: '_pyeval' undefined near line 1 column 1
    Traceback (most recent call last):
      File "site-packages\scipy\io\matlab\mio.py", line 33, in _open_file
    FileNotFoundError: [Errno 2] No such file or directory: 'C:/Users/Jason/AppData/Local/Temp/tmpbn8aefsj/reader.mat'
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "readFolder.py", line 8, in <module>
        octave.eval('pkg load io')
      File "site-packages\oct2py\core.py", line 484, in eval
      File "site-packages\oct2py\core.py", line 369, in feval
      File "site-packages\oct2py\core.py", line 565, in _feval
      File "site-packages\oct2py\io.py", line 28, in read_file
      File "site-packages\scipy\io\matlab\mio.py", line 141, in loadmat
      File "site-packages\scipy\io\matlab\mio.py", line 64, in mat_reader_factory
      File "site-packages\scipy\io\matlab\mio.py", line 39, in _open_file
    FileNotFoundError: [Errno 2] No such file or directory: 'C:/Users/Jason/AppData/Local/Temp/tmpbn8aefsj/reader.mat'
    [1456] Failed to execute script readFolder
```
It appears the issue is first with 'mio.py', but I cannot figure out where it's looking for 'reader.mat'. I think 'reader.mat' is some sort of temporary file created by oct2py.

Just to be sure the python script runs correctly and evaluate the whole oct2py referenced .m file correctly.

Thanks

python 3.7, pyinstaller 3.5 dev, oct2py 4.0.6.

**Comment by blink1073** [comment]
Hi @DoNoRe, that file is supposed to be created by Octave when running `_pyeval.m`, I'm not sure why it would be different when running from an exe.

**Comment by DoNoRe** [comment]
Is there a way to designate where it (reader.mat and even writer.mat) should be made and then set that in stone so the program can find it whether it's run as an executable or a script?

**Comment by blink1073** [comment]
You can set the `temp_dir` property of the session: https://github.com/blink1073/oct2py/blob/ba70aa98453254810c7530c6506234e0f08bc42b/oct2py/core.py#L51

**Comment by kupiqu** [comment]
I get a similar error (error: save: unable to open output file '/tmp/tmp0twzvs0u/reader.mat') using Octave 5.1 via flatpak package in linux.

Somehow writer.mat is there, but reader.mat is not.

**Comment by kupiqu** [comment]
It seems the problem is that the temporal folder in flatpak containers is private to the container. I tried to workaround the issue with a fake octave executable for oct2py that uses the filesystem tmp, but the problem with this is that the process of the fake executable and that of the actual octave process obviously do not match, and oct2py complainsts about that: `OSError: [Errno 8] Exec format error: 'octave4oct2py'`

**Comment by kupiqu** [comment]
"fixed" it with: flatpak override --filesystem=/tmp org.octave.Octave

**Comment by blink1073** [comment]
I don't think there is anything we can do on the `oct2py` end.  We rely on shared files to move the data between `Python` and `Octave`.

**Comment by blink1073** [comment]
I labeled as reference in case someone else has a similar issue.  A PR with some pointers on using `flatpack` in the README would be welcome ;).

**Comment by DoNoRe** [comment]
Sorry about the delay in responding guys..
Also what seems to work is setting:
<<self.temp_dir = "YOUR PRE MADE FOLDER">>

in oct2py/oct2py/core.py per blink1073's answer above. I was able to generate executables successfully using that method.

**Comment by DoNoRe** [comment]
Thanks for the input kupiqu!

**Comment by kupiqu** [comment]
> Sorry about the delay in responding guys..
> Also what seems to work is setting:
> <<self.temp_dir = "YOUR PRE MADE FOLDER">>
> 
> in oct2py/oct2py/core.py per blink1073's answer above. I was able to generate executables successfully using that method.

Cannot you pass that folder to Oct2Py with the temp_dir argument? For instance:

```
oc = Oct2Py(temp_dir="YOUR PRE MADE FOLDER")
```

https://blink1073.github.io/oct2py/source/api.html

**Comment by kiis7** [comment]
I am experiencing the same problem on Windows executable. I tried specifying the temp folder but still caught the error.

**Comment by gonebanana** [comment]
Hey everybody!
I found a solution for myself in Windows, and can tell if it would be helpful for someone.
1. I have added the function  resource_path(relative_path) as shown here 
[https://stackoverflow.com/questions/42251295/file-path-for-pyinstaller-bundled-images]
2. then I set the temp_dir like this
`oct = oct2py.Oct2Py(temp_dir=resource_path("."))`
3.  running a program via PyCharm and it creates all nessesary files - reader.mat, writer.mat. In my situation it appears in the current directory
4. i add them into .spec file by
`datas=[`
`	('writer.mat', '.'),`
`	('reader.mat', '.')`
`], `
5. create exe by
pyinstaller filename.spec

and all works!
